File: reservationms/src/main/java/com/gotogether/reservationms/model/Reservation.java
package com.gotogether.reservationms.model;

import com.gotogether.reservationms.model.enums.Status;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long trajetId;
    private Long passagerId;
    private int placesReservees;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.PENDING;
    private LocalDateTime createdAt;
}

File: gotogether-frontend/src/app/models/reservation.ts
export interface Reservation {
  id: number;
  trajetId: number;
  passagerId: number;
  passagerNom?: string;
  placesReservees: number;
  status: 'PENDING' | 'CONFIRMED' | 'CANCELLED' | 'REFUSED';
  createdAt: string;
}
